### ISSUE TITLE
Library artifacts are published to jcenter only, which will close in 3 months

### ISSUE THREAD

**Issue by entibaldi** [yarolegovich/DiscreteScrollView#204]
Hello! Your library is published only to jcenter. Please consider publishing it also to mavenCentral, as jcenter is [shutting down in May](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/).

[Here](https://central.sonatype.org/pages/producers.html) is some info on how to publish artifacts to mavenCentral.

**Comment by jemshit** [comment]
```Could not HEAD 'http://dl.bintray.com/amulyakhare/maven/com/yarolegovich/discrete-scrollview/1.5.1/discrete-scrollview-1.5.1.pom'. Received status code 403 from server: Forbidden```

no jcenter() in code

Solution?

**Comment by stephanepechard** [comment]
@jemshit It's because the library is not available on any server but jcenter. If you removed it, you cannot have it.
That's the subject of this issue.

**Comment by devaniumesh** [comment]
Please make it available to another repository. jcenter() is deprecated already

**Comment by Ynnck123** [comment]
Any updates on this topic?

**Comment by jemshit** [comment]
I published it on [jitpack](https://github.com/jemshit/DiscreteScrollView), you can do the same, check my last commit. Still, library internally uses jcenter() to pull [another library](https://github.com/yarolegovich/MaterialPreferences/issues/18), but i don't think it is problem, since jitpack will need it while building, once, not afterwards.

**Comment by einschneidend** [comment]
> JCenter will allow downloads of existing artifacts indefinitely

https://developer.android.com/studio/build/jcenter-migration

**Comment by gildor** [comment]
@einschneidend Jcenter is an unreliable service with a bad reputation, recently they had a big outage which break also Gradle Plugin Portal because they still rely on it.
Many companies restrict dependencies from Jcenter (we do the same), we re-published DiscreteScrollView to our own repository but now we plan to remove dependency on DiscreteScrollView because of lack of support

**Comment by avalanchas** [comment]
@gildor which other dependecy are you replacing it with? or are you coding this behaviour yourself now?

**Comment by gildor** [comment]
@avalanchas yes, we already use ViewPager2 in some cases and plan to replace rest of usages but it require more cusomization

**Comment by idrios** [comment]
I'm throwing a comment down because this widget is objectively better than trying to use Android's builtin RecyclerView with SnapHelper. 

To anyone wanting to use this project but unable to because it's not published anymore, the steps you can take are: 

1. Clone this repository
2. Build the project to get the .aar file. Easiest way to do this is to just run 
```
./gradlew assemble
```
The release version of the project will be in 
```
DiscreteScrollView -> library -> build -> outputs -> aar 
```
3. Move library-release.aar from DiscreteScrollView into the libs file of your project, that is 
```
YourProject -> app -> libs
``` 
4. Update your project's build.gradle(app) to recognize .aar files if it doesn't already. So update this line in your dependencies: 
```gradle
dependencies {
    implementation fileTree(dir: "libs", include: ["*.jar","*.aar"])
    ....
}
```
5. Resync gradle. Congrats on your new and improved RecyclerView!

This information will probably be outdated before it has a chance to help anyone.

**Comment by stephanepechard** [comment]
Thanks @idrios for the initial idea. It is even easier if you already have the lib as a dependency, you can just find the `.aar` file in your `.gradle` directory. Not up-to-date, but the lib is not much upgraded anyway.

**Comment by zhongxianyin** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by ttpho** [comment]
If you can not build `arr`  file, you can download it 
```
https://github.com/devapro/mmvn/blob/0675a4f7f0f15838a56451a26b130d252d228fdd/com/outbrain/discrete-scrollview/1.5.1/discrete-scrollview-1.5.1.aar
```

and use it 

- add file into `libs` folder 
`app/libs/discrete-scrollview-1.5.1.aar`

- file `app/build.gradle`, add 
```
implementation(name:'discrete-scrollview-1.5.1', ext:'aar')
```
- file `build.gradle`, add 

```
allprojects {
    repositories {
        // ...
        flatDir {
            dirs("libs")
        }
    }
}
```

**Comment by zhongxianyin** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by ttpho** [comment]
D   Cv

**Comment by zhongxianyin** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by malik7224** [comment]
@ttpho thank you so much brother
